code/game/objects/items/devices/uplinks.dm

```diff
@@ -175,14 +175,14 @@ var/list/world_uplinks = list()
 	return 0
 //Refund proc for the borg teleporter (later I'll make a general refund proc if there is demand for it)
 /obj/item/device/radio/uplink/attackby(obj/item/weapon/W, mob/user, params)
-	if(istype(W, /obj/item/weapon/antag_spawner/nuke_ops))
-		var/obj/item/weapon/antag_spawner/nuke_ops/S = W
-		if(!S.used)
-			hidden_uplink.uses += S.TC_cost
-			qdel(S)
-			user << "<span class='notice'>Teleporter refunded.</span>"
-		else
-			user << "<span class='warning'>This teleporter is already used!</span>"
+	if(istype(W))
+		for(var/datum/uplink_item/D in subtypesof(/datum/uplink_item))
+			if(D.item == W.type && D.refundable)
+				hidden_uplink.uses += D.cost
+				hidden_uplink.used_TC -= D.cost

```

Comment:
Change all of these D vars to use initial(). You're looping through paths, not instances.